docs/metrics.md

```diff
@@ -560,7 +560,82 @@ The exported metrics would be:
   words, attributes used to create `Meter` or `Resource` attributes are not
   subject to this cap.
 
-// TODO: Document how to pick cardinality limit.
+#### Cardinality Limits - How to Choose the Right Limit
+
+Choosing the right cardinality limit is crucial for maintaining efficient memory
+usage and predictable performance in your metrics system. The optimal limit
+depends on your temporality choice and application characteristics.
+
+Setting the limit incorrectly can have consequences:
+
+* **Limit too high**: Due to the SDK's [memory
+  preallocation](#memory-preallocation) strategy, excess memory will be
+  allocated upfront and remain unused, leading to resource waste.
+* **Limit too low**: Measurements will be folded into the overflow bucket
+  (`{"otel.metric.overflow": true}`), losing granular attribute information and
+  making attribute-based queries unreliable.
+
+Consider these guidelines when determining the appropriate limit:
+
+**1. For Cumulative Temporality:**
+
+* Cumulative metrics retain their state across export cycles, so the total
+  cardinality equals the Cartesian product of all attribute combinations.
```

Comment:
We want the count of items in the Cartesian product not the product itself.

color = {"red", "green", "yellow"}
name = { "apple", banana"}

Cartesian product = {("red", "apple"), ('green", "apple"), ...}

```suggestion
  cardinality equals the cardinality of the Cartesian product of all attribute sets.
```

Reply:
I think this would be more accurate:

``so the total cardinality equals the size of the Cartesian product of all attribute combinations seen so far.``

or even simpler

``so the total cardinality equals the number of unique attribute value combinations observed so far.``